File: ts/src/modules/comment.ts
/** Comments and threaded reply moderation endpoints */

export class CommentModule {
  private client: any;

  constructor(client: any) {
    this.client = client;
  }

  public async list(mediaId: string | number, minId?: string): Promise<any> {
    const params: Record<string, string> = {};
    if (minId) params.min_id = minId;
    return this.client.get(`/api/v1/media/${mediaId}/comments/`, Object.keys(params).length ? params : undefined);
  }

  public async replies(mediaId: string | number, commentId: string | number, minId?: string): Promise<any> {
    const params: Record<string, string> = {};
    if (minId) params.min_id = minId;
    return this.client.get(
      `/api/v1/media/${mediaId}/comments/${commentId}/child_comments/`,
      Object.keys(params).length ? params : undefined
    );
  }

  public async add(
    mediaId: string | number,
    text: string,
    repliedToCommentId?: string | number
  ): Promise<any> {
    const data: Record<string, string> = { comment_text: text };
    if (repliedToCommentId) data.replied_to_comment_id = String(repliedToCommentId);
    return this.client.post(`/api/v1/media/${mediaId}/comment/`, data);
  }

  public async like(commentId: string | number): Promise<any> {
    return this.client.post(`/api/v1/media/${commentId}/comment_like/`);
  }

  public async unlike(commentId: string | number): Promise<any> {
    return this.client.post(`/api/v1/media/${commentId}/comment_unlike/`);
  }

  public async delete(mediaId: string | number, commentId: string | number): Promise<any> {
    return this.client.post(`/api/v1/media/${mediaId}/comment/${commentId}/delete/`);
  }

  public async pin(mediaId: string | number, commentId: string | number): Promise<any> {
    return this.client.post(`/api/v1/media/${mediaId}/comment/${commentId}/pin/`);
  }

  public async unpin(mediaId: string | number, commentId: string | number): Promise<any> {
    return this.client.post(`/api/v1/media/${mediaId}/comment/${commentId}/unpin/`);
  }
}

File: instagramflow/modules/comment.py
"""Comment engagement and moderation endpoints."""
from __future__ import annotations

class CommentModule:
    """Operations on post and Reels comments."""

    def __init__(self, client):
        self._client = client

    def list(self, media_id: str | int, min_id: str | None = None) -> dict:
        """Fetch comments for a media post or reel."""
        params = {}
        if min_id:
            params["min_id"] = min_id
        return self._client._get(f"/api/v1/media/{media_id}/comments/", params=params or None)

    def replies(self, media_id: str | int, comment_id: str | int, min_id: str | None = None) -> dict:
        """Fetch child comment replies under a specific parent comment."""
        params = {}
        if min_id:
            params["min_id"] = min_id
        return self._client._get(
            f"/api/v1/media/{media_id}/comments/{comment_id}/child_comments/",
            params=params or None,
        )

    def add(
        self,
        media_id: str | int,
        text: str,
        replied_to_comment_id: str | int | None = None,
    ) -> dict:
        """Post a comment or reply to an existing comment on a media item."""
        data = {"comment_text": text}
        if replied_to_comment_id:
            data["replied_to_comment_id"] = str(replied_to_comment_id)
        return self._client._post(f"/api/v1/media/{media_id}/comment/", data=data)

    def like(self, comment_id: str | int) -> dict:
        """Like a specific comment."""
        return self._client._post(f"/api/v1/media/{comment_id}/comment_like/")

    def unlike(self, comment_id: str | int) -> dict:
        """Unlike a specific comment."""
        return self._client._post(f"/api/v1/media/{comment_id}/comment_unlike/")

    def delete(self, media_id: str | int, comment_id: str | int) -> dict:
        """Delete a comment posted by the current account."""
        return self._client._post(f"/api/v1/media/{media_id}/comment/{comment_id}/delete/")

    def pin(self, media_id: str | int, comment_id: str | int) -> dict:
        """Pin a comment to the top of the comments thread."""
        return self._client._post(f"/api/v1/media/{media_id}/comment/{comment_id}/pin/")

    def unpin(self, media_id: str | int, comment_id: str | int) -> dict:
        """Unpin a previously pinned comment."""
        return self._client._post(f"/api/v1/media/{media_id}/comment/{comment_id}/unpin/")
